tools/lib/git.source

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+#
+#    Copyright © 2016 The qTox Project Contributors
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# source for scripts using git-based functionality
+
+set -e -o pipefail
+
+# Get numerical version from git tag, parts of version separated by dots
+# from e.g. `v123.456.789-321-asdf` get `123.456.789` part
+get_version() {
+    git describe --tags \
```

Comment:
You could use `git describe --abbrev=0` instead of `git describe --tags` to get 'v123.456.789' directly, and then you can just strip the leading "v" if you want.